synapseclient/__main__.py

```diff
@@ -134,7 +134,8 @@ def store(args, syn):
     # If both args.FILE and args.file specified raise error
     if args.file and args.FILE:
         raise ValueError('only specify one file')
-
+    if args.type == 'File' and not args.file and not args.FILE:
+        raise ValueError(f'For the {args.subparser} command, put the positional FILE as the first arg')
```

Comment:
I think this error message is a bit too specific. We could maybe just just say
"{args.subpaser} missing required FILE argument"